### DISCUSSION TITLE
How to cut by a circle in a two-dimensional model？

### DISCUSSION THREAD

**Discussion by bright-men** [neperfepx/neper#826] [Q&A] [answered]
There is a case of 3D digging in the help document.
neper -T -n 1000 -morpho gg -domain "cube(1.2,0.1,1.2)" -transform "cut(cylinder(0.3,0,0.3,0,1,0,0.15),cylinder(0.3,0,0.9,0,1,0,0.15),cylinder(0.9,0,0.3,0,1,0,0.15),cylinder(0.9,0,0.9,0,1,0,0.15))"

I want to cut by a circle in a 2D model, can this be done in neper？Consider the diagram below
![2024-03-12_181716](https://github.com/neperfepx/neper/assets/56397035/b12d08c0-ba3f-40c8-b595-4d8c5226a335)

**Comment by rquey** [comment, marked as answer]
```
neper -T -n 1000 -dim 2 -domain "square(1.2,1.2)" -transform "cut(cylinder(0.3,0.3,0,0,0,1,0.15),cylinder(0.3,0.9,0,0,0,1,0.15),cylinder(0.9,0.3,0,0,0,1,0.15),cylinder(0.9,0.9,0,0,0,1,0.15))"
```
![d](https://github.com/neperfepx/neper/assets/2659376/2c6f3b2e-4b01-4091-a4ff-bf0fe4ec8b30)

**Reply by bright-men** [reply]
Many thanks!
